transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -722,7 +708,7 @@ private void invokeWrite(Object msg, ChannelPromise promise) {
         try {
             handler().write(this, m, promise);
         } catch (Throwable t) {
-            notifyOutboundHandlerException(t, promise);
+            handleOutboundHandlerException(t);
```

Comment:
Might be a dumb question, but how does the promise get failed incase there is a problem?

Reply:
it doesn't... We could call `promise.tryFailure(...)` tho. Not sure if we should or not ... @NiteshKant @chrisvest WDYT ?

Reply:
Let me do this...

Reply:
@carl-mastrangelo done in 78a1912

Reply:
`tryFailure()` makes sense.